### ISSUE TITLE
cv2.VideoCapture(0) does not work

### ISSUE THREAD

**Issue by timkay** [pyodide/pyodide#4315]
## 🐛 Bug

cv2.VideoCapture(0) does not work

### To Reproduce

In https://pyodide.org/en/stable/console.html, type
```
import cv2
cap = cv2.VideoCapture(0)
cap.isOpened()
cap.read()
```

I get

![image](https://github.com/pyodide/pyodide/assets/65144/c4bffe89-9172-45d1-966d-da05e07aead6)

I have tried with webcam 1, 2, 3 as well, with the same result.

### Expected behavior

`cap.isOpened()` should return `True`, and `cap.read()` should return `(True, frame)`.

### Environment

- Pyodide Version 0.24.1
- Browser version Chrome 119.0.6045.159 and Edge 121.0.2220.3
- Any other relevant information:

**Comment by ryanking13** [comment]
Yeah, it would be great to have some function to take an object from a web API and mount it to a virtual file system.

This would require a low-level implementation, so probably it needs to be done in Emscripten.

**Comment by hoodmane** [comment]
I think you can use `FS.createDevice` for this sort of thing.

**Comment by pmp-p** [comment]
pygame-web has already some support for [webcams](https://github.com/pygame-web/platform_wasm/blob/main/platform_wasm/pygame/vidcap.py), it is only a few lines of javascript glue, an offscreen canvas and a fake /dev/video0 ( it is a png file ). You would just nead to monkey patch the read()  function of opencv capture with a format converter png/jpeg => opencv format.

sidenote: webcams work only for localhost and https url contexts.
